pyanaconda/ui/gui/spokes/advstorage/zfcp.py

```diff
@@ -89,53 +94,70 @@ def on_start_clicked(self, *args):
         self._startButton.hide()
         self._cancelButton.set_sensitive(False)
         self._okButton.set_sensitive(False)
-
-        self._conditionNotebook.set_current_page(1)
         self._set_configure_sensitive(False)
-        self._deviceEntry.set_sensitive(False)
-
         self._conditionNotebook.set_current_page(1)
-        # below really, really is ugly and needs to be re-factored, but this
-        # should give a good base idea as far as expected behavior should go
-        try:
-            device = blockdev.s390.sanitize_dev_input(self._deviceEntry.get_text())
-            wwpn = blockdev.s390.zfcp_sanitize_wwpn_input(self._wwpnEntry.get_text())
-            lun = blockdev.s390.zfcp_sanitize_lun_input(self._lunEntry.get_text())
-        except blockdev.S390Error as err:
-            _config_error = str(err)
-            self.builder.get_object("deviceErrorLabel").set_text(_config_error)
-            self._conditionNotebook.set_current_page(2)
 
-        spinner = self.builder.get_object("waitSpinner")
-        spinner.start()
-
-        self._discover(device, wwpn, lun)
-        self._check_discover()
+        # Initialize.
+        config_error = None
+        device = None
+        wwpn = None
+        lun = None
+
+        # Get the input.
+        device_name = self._deviceEntry.get_text().strip()
+        wwpn_name = self._wwpnEntry.get_text().strip()
+        lun_name = self._lunEntry.get_text().strip()
+
+        # Check the input.
+        if not DASD_DEVICE_NUMBER.match(device_name):
+            config_error = "Incorrect format of the given device number."
+        elif not ZFCP_WWPN_NUMBER.match(wwpn_name):
+            config_error = "Incorrect format of the given WWPN number."
+        elif not ZFCP_LUN_NUMBER.match(lun_name):
+            config_error = "Incorrect format of the given LUN number."
+        else:
+            try:
+                # Get the full ids.
+                device = blockdev.s390.sanitize_dev_input(device_name)
+                wwpn = blockdev.s390.zfcp_sanitize_wwpn_input(wwpn_name)
+                lun = blockdev.s390.zfcp_sanitize_lun_input(lun_name)
+            except (blockdev.S390Error, ValueError) as err:
+                config_error = str(err)
+
+        # Process the configuration error.
+        if config_error:
+            self._errorLabel.set_text(config_error)
+            self._conditionNotebook.set_current_page(2)
+            self._set_configure_sensitive(True)
+            self._cancelButton.set_sensitive(True)
+        # Start the discovery.
+        else:
+            self._spinner.start()
```

Comment:
I'm not sure but is this working? 
Isn't the whole UI blocked because of the `_discover()` method which means the `spiner` won't be animating? Maybe I'm wrong and the spinner works in a separate thread, however I fear it is not.